compiler/passes/convert-uast.cpp

```diff
@@ -952,16 +996,23 @@ struct Converter {
     bool maybeArrayType = isLoopMaybeArrayType(node);
     bool zippered = node->iterand()->isZip();
 
-      // Unpack things differently if body is a conditional.
-      if (auto origCond = node->stmt(0)->toConditional()) {
-        INT_ASSERT(origCond->numThenStmts() == 1);
-        INT_ASSERT(!origCond->hasElseBlock());
-        expr = singleExprFromStmts(origCond->thenStmts());
+    // Unpack things differently if body is a conditional.
+    if (auto origCond = node->stmt(0)->toConditional()) {
+      INT_ASSERT(origCond->numThenStmts() == 1);
+      // a filter expression
+      if (!origCond->hasElseBlock()) {
         cond = convertAST(origCond->condition());
+        expr = singleExprFromStmts(origCond->thenStmts());
         INT_ASSERT(cond);
       } else {
-        expr = singleExprFromStmts(node->stmts());
+        // an assignment expression
```

Comment:
Not sure why it says "an assignment expression". Is it supposed to say it is a regular conditional expression in the body?

Reply:
I meant to differentiate between a filter and an assignment in the loop body. So something like this:
```
const A: [1..10] int = [i in 1..10] if (i < 3) then 100 else i;
``` 
where the expression is an assignment at the indices of the loop instead of:
```
[i in A] if (i > 3) then writeln(i);
``` 
where the expression filters on the value

Reply:
right but 

> `const A: [1..10] int = [i in 1..10] if (i < 3) then 100 else i;`

does not have an assignment in the loop body (it has a conditional in the loop body and the assignment is outside of the loop expression)

e.g. you could have

> `var A = [i in A] if (i > 3) then writeln(i);`

which is also an assignment but still a filter